samtranslator/utils/traverse.py

```diff
@@ -0,0 +1,104 @@
+from typing import Any, Dict, List
+
+from samtranslator.utils.resolve_actions import ResolveAction
+
+
+def traverse(template: Dict[str, Any], actions: List[ResolveAction]) -> Any:
+    """
+    Calls the correct traversal method
+
+    :param template: The template that needs modifying
+    :param actions: That actions that need to be performed to the template
+    :return: Modified Template
+
+    TODO: Add multiple strategies and choose between those with an Optional value
+    """
+    return _preorder_traverse(template, actions)
```

Comment:
There is no need to have separate function for pre-order.
Since we have > 2 leaves, in-order is not possible and all we possibly can have is pre-order and post-order. The difference is only in applying actions before traversing children or after. It can be implemented by have 2 different actions parameters `preorder_actions` and `postorder_actions` instead of `actions`. But I suppose that postorder would require some data from children, otherwise, why would we care about applying actions post-factum. Since we don't  have a use case now, instead of getting too abstract, lets implement preorder only, right in the `travers()` function